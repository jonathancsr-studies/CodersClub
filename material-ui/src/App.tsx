import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { Button, Typography } from '@mui/material'

function App() {
  const [count, setCount] = useState(0)

  return (
    <div className="App">
      <Typography
         variant='h6'
      >
        Hello From Mui
      </Typography>
  
      <Button variant='contained'>
          Hello From Mui
      </Button>
    </div>
  )
}

export default App
