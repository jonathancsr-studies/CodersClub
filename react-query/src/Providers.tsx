import React, { PropsWithChildren } from "react";
import { QueryClientProvider } from "react-query";
import queryClient from "./services/queryClient";

export const Providers: React.FC<PropsWithChildren> = ({ children }) => {
  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};
