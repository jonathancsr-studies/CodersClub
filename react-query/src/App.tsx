import { useCallback, useState } from "react";
import "./App.css";
import useFetchTopMangas from "./requests/useFetchTopMangas";
import { useFavoriteMangas } from "./store/useFavoriteMangas";

function App() {
  const [showOnlyFavorites, setShowOnlyFavorites] = useState(false);

  const { data } = useFetchTopMangas();
  const { favorites, favoritateManga } = useFavoriteMangas();

  const toggleShowOnlyFavorites = useCallback(() => {
    setShowOnlyFavorites((prevShowOnlyFavorites) => !prevShowOnlyFavorites);
  }, [favorites]);
  return (
    <div className="App" style={{ display: "flex", flexWrap: "wrap" }}>
      <button onClick={toggleShowOnlyFavorites}>Show Favorites</button>
      {data?.data
        .filter(
          (manga) => !showOnlyFavorites || favorites.includes(manga.mal_id)
        )
        .map((manga) => (
          <div
            key={manga.mal_id}
            style={{ display: "flex", flexDirection: "column" }}
          >
            {manga.title}
            <img src={manga.images.jpg.image_url} alt="" />
            <button
              onClick={() => favoritateManga(manga.mal_id)}
              data-testid={`Favorite-Button-${manga.mal_id}`}
            >
              {favorites.includes(manga.mal_id) ? "*" : ""} Favoritar
            </button>
          </div>
        ))}
    </div>
  );
}

export default App;
