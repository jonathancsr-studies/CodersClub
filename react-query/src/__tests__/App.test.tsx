import { render, screen, waitFor } from "@testing-library/react";
import nock from "nock";
import App from "../App";
import { Providers } from "../Providers";
import { topMangas } from "../__mocks__/topMangas";

nock("https://api.jikan.moe/v4")
  .persist()
  .defaultReplyHeaders({
    "access-control-allow-origin": "*",
  })
  .get("/top/manga")
  .reply(200, topMangas);

test("renders mangas successfully", async () => {
  render(
    <Providers>
      <App />
    </Providers>
  );
  const firstTopManga = topMangas.data[0];

  await waitFor(() => {
    const firstElement = screen.getByText(firstTopManga.title);
    return expect(firstElement).toBeInTheDocument();
  });
});

test("favoritate manga successfully", async () => {
  render(
    <Providers>
      <App />
    </Providers>
  );
  const firstTopManga = topMangas.data[0];
  const secondTopManga = topMangas.data[1];

  const favoritateFirstMangaButton = screen.getByTestId(
    `Favorite-Button-${firstTopManga.mal_id}`
  );
  const showFavoriteMangasButton = screen.getByText(/Show Favorites/i);

  favoritateFirstMangaButton.click();
  showFavoriteMangasButton.click();

  await waitFor(() => {
    const firstMangaElement = screen.getByText(firstTopManga.title);
    expect(firstMangaElement).toBeInTheDocument();
  });

  await waitFor(() => {
    const secondMangaElement = screen.queryByText(secondTopManga.title);
    expect(secondMangaElement).not.toBeInTheDocument();
  });
});
it("Match Snapshot", async () => {
  const { container } = render(
    <Providers>
      <App />
    </Providers>
  );
  await waitFor(() => {
    expect(container).toMatchSnapshot();
  });
});
