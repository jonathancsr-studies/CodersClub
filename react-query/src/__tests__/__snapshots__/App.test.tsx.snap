// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot 1`] = `
<div>
  <div
    class="App"
    style="display: flex; flex-wrap: wrap;"
  >
    <button>
      Show Favorites
    </button>
    <div
      style="display: flex; flex-direction: column;"
    >
      Berserk
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/1/157897.jpg"
      />
      <button
        data-testid="Favorite-Button-2"
      >
        *
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      JoJo no Kimyou na Bouken Part 7: Steel Ball Run
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/179882.jpg"
      />
      <button
        data-testid="Favorite-Button-1706"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Vagabond
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/1/259070.jpg"
      />
      <button
        data-testid="Favorite-Button-656"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      One Piece
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/253146.jpg"
      />
      <button
        data-testid="Favorite-Button-13"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Monster
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/258224.jpg"
      />
      <button
        data-testid="Favorite-Button-1"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Slam Dunk
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/258749.jpg"
      />
      <button
        data-testid="Favorite-Button-51"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Fullmetal Alchemist
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/243675.jpg"
      />
      <button
        data-testid="Favorite-Button-25"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Grand Blue
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/166124.jpg"
      />
      <button
        data-testid="Favorite-Button-70345"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Vinland Saga
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/188925.jpg"
      />
      <button
        data-testid="Favorite-Button-642"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Oyasumi Punpun
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/266834.jpg"
      />
      <button
        data-testid="Favorite-Button-4632"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Kingdom
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/171872.jpg"
      />
      <button
        data-testid="Favorite-Button-16765"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Houseki no Kuni
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/1/115443.jpg"
      />
      <button
        data-testid="Favorite-Button-44489"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      20th Century Boys
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/5/260006.jpg"
      />
      <button
        data-testid="Favorite-Button-3"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Ashita no Joe
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/1/268827.jpg"
      />
      <button
        data-testid="Favorite-Button-1303"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Monogatari Series: First Season
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/5/173535.jpg"
      />
      <button
        data-testid="Favorite-Button-14893"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Real
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/115969.jpg"
      />
      <button
        data-testid="Favorite-Button-657"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Kaguya-sama wa Kokurasetai: Tensai-tachi no Renai Zunousen
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/188896.jpg"
      />
      <button
        data-testid="Favorite-Button-90125"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Youkoso Jitsuryoku Shijou Shugi no Kyoushitsu e
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/177958.jpg"
      />
      <button
        data-testid="Favorite-Button-89357"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Monogatari Series: Second Season
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/181553.jpg"
      />
      <button
        data-testid="Favorite-Button-23751"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Youkoso Jitsuryoku Shijou Shugi no Kyoushitsu e: 2-nensei-hen
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/235363.jpg"
      />
      <button
        data-testid="Favorite-Button-123992"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Mikkakan no Koufuku
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/3/248674.jpg"
      />
      <button
        data-testid="Favorite-Button-126479"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      GTO
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/172982.jpg"
      />
      <button
        data-testid="Favorite-Button-336"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      3-gatsu no Lion
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/2/52283.jpg"
      />
      <button
        data-testid="Favorite-Button-1224"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Yotsuba to!
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/5/259524.jpg"
      />
      <button
        data-testid="Favorite-Button-104"
      >
         Favoritar
      </button>
    </div>
    <div
      style="display: flex; flex-direction: column;"
    >
      Monogatari Series: Final Season
      <img
        alt=""
        src="https://cdn.myanimelist.net/images/manga/1/181554.jpg"
      />
      <button
        data-testid="Favorite-Button-44227"
      >
         Favoritar
      </button>
    </div>
  </div>
</div>
`;
